types/tx_msg.go

```diff
@@ -80,6 +80,12 @@ type (
 		GetMemo() string
 	}
 
+	// MsgWithMoniker must have GetMoniker() method to use CollectTx
```

Comment:
Can we have this in x/gentuil instead?
I wouldn't make this interface public actually

Reply:
So where should we put it?

Reply:
just inline in the interface check in genutil imho